packages/core/test/helpers/ast.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { expect } from 'chai';
+import { parse } from 'postcss';
+import { getAstNodeAt } from '@stylable/core/dist/helpers/ast';
+import deindent from 'deindent';
+
+function setupWithCursor(source: string) {
+    const deindented = deindent(source);
+
+    const position = deindented.indexOf(`|`);
+    return {
+        position,
+        ast: parse(deindented.split(`|`).join(``)),
+    };
+}
+
+// ToDo: make added syntax zero space: semicolon at declaration end
+describe('helpers/ast', () => {
+    describe(`getAstNodeAt`, () => {
+        it(`should find position in declaration property (start)`, () => {
+            const { position, ast } = setupWithCursor(`
+                .selector {
+                    |decl: declValue;
+                }
+            `);
+
+            const [node, offsetInNode] = getAstNodeAt(ast, position);
+
+            expect(node, 'node').to.equal((ast as any).nodes[0].nodes[0]);
+            expect(offsetInNode, 'offset').to.equal(0);
+        });
+        it(`should find position in declaration value (end)`, () => {
+            const { position, ast } = setupWithCursor(`
+                .selector {
+                    decl1: declValue|;decl2: otherValue;
+                }
+            `);
+
+            const [node, offsetInNode] = getAstNodeAt(ast, position);
+            // ToDo: change to return value ast node
+            expect(node, 'node').to.equal((ast as any).nodes[0].nodes[0]);
+            expect(offsetInNode, 'offset').to.equal(16);
+        });
+        it(`should find position in declaration property (end)`, () => {
+            const { position, ast } = setupWithCursor(`
+                .selector {
+                    decl|: declValue;
+                }
+            `);
+
+            const [node, offsetInNode] = getAstNodeAt(ast, position);
+
+            expect(node, 'node').to.equal((ast as any).nodes[0].nodes[0]);
+            expect(offsetInNode, 'offset').to.equal(4);
+        });
+        it(`should find position in rule body (start)`, () => {
+            const { position, ast } = setupWithCursor(`
+                .selector {|
+                    decl: declValue;
+                }
+            `);
+
+            const [node, offsetInNode] = getAstNodeAt(ast, position);
+
+            expect(node, 'node').to.equal((ast as any).nodes[0]);
+            expect(offsetInNode, 'offset').to.equal(11);
+        });
+        it(`should find position in rule body (before declaration)`, () => {
+            const { position, ast } = setupWithCursor(`
+                .selector {
+                   | decl: declValue;
+                }
+            `);
+
+            const [node, offsetInNode] = getAstNodeAt(ast, position);
+
+            expect(node, 'node').to.equal((ast as any).nodes[0]);
+            expect(offsetInNode, 'offset').to.equal(15);
+        });
+    });
+
+    xdescribe(`parseForInspection (safe parse)`, () => {
```

Comment:
Remove?

Reply:
we left these here intentionally for future work.